File: 001_TwoSum/twosum.py
"""
Problem : Two Sum (https://leetcode.com/problems/two-sum/description/)
"""
class Solution(object):
    """Solution Class"""
    @staticmethod
    def twoSum(nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        num_dict = dict()
        for i, value in enumerate(nums):
            num = target - value
            if num in num_dict:
                return [num_dict[num], i]
            else:
                num_dict[value] = i

        return []

File: 001_TwoSum/twosum.cpp
/*
Problem : Two Sum (https://leetcode.com/problems/two-sum/description/)
*/
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> results;
        
        int vecSize = nums.size();
        for (int i = 0; i < vecSize; ++i)
        {
            for (int j = i + 1; j < vecSize; ++j)
            {
                if (nums[i] + nums[j] == target)
                {
                    results.push_back(i);
                    results.push_back(j);
                }
            }
        }

        return results;
    }
};